src/kvstore/wal/test/FileBasedWalTest.cpp

```diff
@@ -387,6 +387,99 @@ TEST(FileBasedWal, RollbackToZero) {
     ASSERT_EQ(0, wal->lastLogId());
 }
 
+TEST(FileBasedWal, TTLTest) {
+    FileBasedWalPolicy policy;
+    TempDir walDir("/tmp/testWal.XXXXXX");
+
+    policy.ttl = 3;
+    policy.bufferSize = 128;
+    policy.fileSize = 1024;
+    policy.numBuffers = 30;
+    // Create a new WAL, add one log and close it
+    auto wal = FileBasedWal::getWal(walDir.path(),
+                                    policy,
+                                    flusher.get(),
+                                    [](LogID, TermID, ClusterID, const std::string&) {
+                                        return true;
+                                    });
+    EXPECT_EQ(0, wal->lastLogId());
+
+    for (int i = 1; i <= 1000; i++) {
+        EXPECT_TRUE(
+            wal->appendLog(i /*id*/, 1 /*term*/, 0 /*cluster*/,
+                           folly::stringPrintf("Test string %02d", i)));
+    }
+    sleep(1);
+    auto startIdAfterGC
+        = static_cast<FileBasedWal*>(wal.get())->walFiles_.rbegin()->second->firstId();
+    auto expiredFilesNum = static_cast<FileBasedWal*>(wal.get())->walFiles_.size();
```

Comment:
`auto expiredFilesNum = static_cast<FileBasedWal*>(wal.get())->walFiles_.size() - 1;` is better.

Reply:
Good idea